### ISSUE TITLE
`Fetcher` with `?index` route has the URL parameter overridden

### ISSUE THREAD

**Issue by huw** [remix-run/remix#4221]
### What version of Remix are you using?

1.7.1

### Steps to Reproduce

Create a manual `Fetcher`, perhaps using `useFetcher`. Then submit it against the index route using a GET request.

```ts
const fetcher = useFetcher();
fetcher.submit({ parameter: “true” }, { replace: false, method: “get”, action: “?index” });
```

(I’m calling this from the index route, so I think you can omit all of these options. But worth being verbose to explain the error.)

### Expected Behavior

This code should call the index loader & return its data in `fetcher.data`

### Actual Behavior

`fetcher.data` is undefined

([this example in the docs is now broken, BTW](https://remix.run/docs/en/v1/api/remix#fetchersubmit))

### Reason

In #4046, we override the URL parameters for a GET request. Since `Fetcher` uses this under the hood, the `?index` search string [gets wiped from the URL](https://github.com/remix-run/remix/blob/b35e1355d27353c1946c56842be992bbc5d1d60c/packages/remix-react/components.tsx#L1224). Then when calling `getFetcherRequestMatch`, [`match.route.index` is no longer true](https://github.com/remix-run/remix/blob/b35e1355d27353c1946c56842be992bbc5d1d60c/packages/remix-react/transition.ts#L639), so it can’t find the index route anymore, so it calls the default fetcher, which silently returns `undefined`.

(I also happened to be relying on the old behaviour by submitting `action: “?index&parameter=true”` which broke in the new version, but I didn’t realise you could use the first parameter. Only mentioning in case someone else has the same problem.)

### Workaround

You have to downgrade to 1.7.0. I tried:

```ts
fetcher.submit({ parameter: “true”, index: “anything” }, options)
```

But this doesn’t work because [it only matches against a bare `index` parameter](https://github.com/remix-run/remix/blob/b35e1355d27353c1946c56842be992bbc5d1d60c/packages/remix-react/transition.ts#L620), which I can’t pass using this object.

**Comment by brophdawg11** [comment]
This was a regression on our part - normal form GET submissions run all loaders and thus don't need the `?index` param, but this `fetcher.submit({}, { method: 'get' })` case was overlooked.  Sorry about that!  Fixed in #4238 and should be released soon 👍

**Comment by MichaelDeBoey** [comment]
Closed by #4238
